ib.c.ib

```diff
@@ -109,8 +109,9 @@ static size_t get_spaces(const size_t tab)
 	return spaces == 0 ? tab : tab * spaces
 
 static void strip_newline(char *str)
-	if (str[strlen(str) - 1] == '\n')
-		str[strlen(str) - 1] = '\0'
+	size_t length = strlen(str)
+	if (length > 0 && str[length - 1] == '\n')
+		str[length - 1] = '\0'
```

Comment:
Multiple calls to `strlen` avoided, length might be `0`, so `str[strlen(str) - 1]` might be invalid.